pkg/cmd/api/fields_test.go

```diff
@@ -61,8 +61,14 @@ func Test_parseFields_nested(t *testing.T) {
 			"robots[]=Dependabot",
 			"labels[][name]=bug",
 			"labels[][color]=red",
+			"labels[][colorOptions][]=red",
+			"labels[][colorOptions][]=blue",
 			"labels[][name]=feature",
 			"labels[][color]=green",
+			"labels[][colorOptions][]=red",
+			"labels[][colorOptions][]=green",
+			"labels[][colorOptions][]=yellow",
+			"nested[][key1][key2][key3]=value",
```

Comment:
This is a test case that worked before the changes in this PR, but was missing and should probably be here to cover our bases

Reply:
**What are the various error conditions this should fail and what should `gh` show when these appear?**

Using `smee` to test out permutations against a local `make` build, one example that failed:

```shell
$ ./bin/gh api "https://smee.io/..." -F 'nested_property[mixed][]=1' -F 'nested_property[mixed][foo]=bar'
expected map type under "mixed", got []interface {}
```

So, this erred as expected but it would help to have a more specific error message explaining why this fails.  Also the inverse would likely fail and should explain the reverse:

```shell
$ ./bin/gh api "https://smee.io/pkeWXMdpNhNWI5t" \
        -F 'nested_property[mixed][foo]=bar' \
        -F 'nested_property[mixed][]=1'

expected array type under "mixed", got map[string]interface {}
```

Reply:
_I realize this is commentary about existing behavior outside of your change, so not blocking._